core/commonMain/src/kotlinx/io/text/TextOutputObsolete.kt

```diff
@@ -11,16 +10,16 @@ private const val lastASCII = 0x7F.toChar()
  *
  * @throws MalformedInputException if encoding is invalid.
  */
-fun Output.writeUtf8String(text: CharSequence, index: Int = 0, length: Int = text.length - index) {
+public fun Output.writeUtf8String(text: CharSequence, index: Int = 0, length: Int = text.length - index) {
     var textIndex = index // index in text
     val textEndIndex = index + length // index of char after last
     var bytes = 0 // current left bytes to write (in inverted order)
 
     while (textIndex < textEndIndex || bytes != 0) {
-        writeBufferRange { buffer, startOffset, endOffset ->
-            var offset = startOffset
+        writeBuffer { buffer, startIndex, endIndex ->
```

Comment:
This change should be in a standalone commit, with corresponding tests and explanation

Reply:
This change is code refactoring: migrating to exclusive `endIndex` instead of inclusive `endOffset` and word `Range` remove. It has no logic changes.